src/IO/FileEncryptionCommon.cpp

```diff
@@ -84,10 +85,13 @@ namespace
         while (in_size < size)
         {
             out.nextIfAtEnd();
+
             size_t part_size = std::min(size - in_size, out.available());
+            part_size = std::min<size_t>(part_size, INT_MAX);
```

Comment:
Should that not throw an exception instead of silently truncate? (--> safe_cast)

Reply:
This function has a loop over input, so should safe.

Reply:
Hm, okay.